### ISSUE TITLE
Error writing to `/etc/iproute2/rt_tables` on NixOS

### ISSUE THREAD

**Issue by 06kellyjac** [mullvad/mullvadvpn-app#1891]
## Issue report

Logs submitted through the report feature. The message mentions the title of this issue.

Operating system:

 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.49, NixOS, 20.09pre232722.b3251e04ee4 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.6`
 - channels(root): `"home-manager, nixos-20.09pre232722.b3251e04ee4, nixos-hardware"`
 - channels(user): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

App version:

```
mullvad --version
mullvad 2020.5
```

### Issue description

See the following issue for more details:

https://github.com/NixOS/nixpkgs/issues/91923

TL;DR, It's trying to write to `/etc/iproute2/rt_tables` which doesn't exist (and probably shouldn't exist??) on my NixOS machine

**Comment by 06kellyjac** [comment]
Adding the line `networking.iproute2.enable = true;` to your nixos config will copy writable configuration files to `/etc/iproute2`
[details](https://github.com/NixOS/nixpkgs/blob/0fa04d646d65152d70e32f75d5f2a9518dce176e/nixos/modules/config/iproute2.nix) so now the mullvad daemon is working.

My question from here would be is it strictly necessary to write to the `rt_tables` file?

If it is then we can have our mullvad service enable this setting.

**Comment by pinkisemils** [comment]
The `rt_tables` are required to get split tunelling to work, and it's easier to just fail quick than to have the split tunelling be configurable dynamically, but we may revise this decision. The `rt_tables` file is required to create new routing tables for a different control group so that processes in that control group. This should've been mentioned in the changelog, but we left it out since this wasn't a _user-facing_ change yet, as we've not advertised it as such, it's only available on Linux and through the CLI. Seemingly, it's more user facing than we expected.

As for a more ideal solution for NixOS, you'd probably need to have a patched version of the daemon to not create new routing tables, and have the routing table be declared by NixOS. Ideally, this app would be more of a service than a package, and enabling it would configure the system to do all of the above. Since we currently have no plans to support NixOS officially, we can't make any promises that we'll be implementing any of the above.

 I too am a NixOS user, and sort of worked through the issue on my own machine the exact same way. I'm using the app in a somewhat weird way (just running the daemon in tmux and such) as I am developing it, so I've not really tested the nixpkgs repackaged version.

**Comment by 06kellyjac** [comment]
Thanks for the response

Not providing an official patch is totally reasonable.
I don't have much experience with networking configuration with NixOS so I'll be sticking to my fix for now, but your solution sounds more fitting.

Once it's patched out:
Would it use `networking.interfaces.<name?>.ipv4.routes` ???
Or would adding `mullvad_exclusions` to `networking.iproute2.rttablesExtraConfig` be  enough? or rewriting `networking.iproute2` to be more declarative?

---

I guess if the mullvad nix package was written to build from source rather than using the `.deb` it'd be easier to point the src at your dev copy
(I guess this would also be necessary for the patching solution :thinking:)

---

Shall we mark this as closed and have further discussion go in `nixos/nixpkgs` ?
